redaxo/src/addons/structure/functions/function_rex_url.php

```diff
@@ -57,10 +57,21 @@ function rex_getUrl($id = null, $clang = null, array $params = [], $separator =
 /**
  * Leitet auf einen anderen Artikel weiter.
  *
+ * @param int       $article_id
+ * @param null|int  $clang
+ * @param array     $params
+ *
+ * @throws InvalidArgumentException
+ *
  * @package redaxo\structure
  */
 function rex_redirect($article_id, $clang = null, array $params = [])
 {
+    // validate $article_id contain int id
+    if ((int)$article_id == $article_id) {
+        throw new InvalidArgumentException(sprintf('Expecting $article_id must contain integer rex_article_id, %s is not an id!', $article_id));
```

Comment:
Was eine "id" ist ist in jedem context anders. 

Vorschlag: `%s is not a valid article_id`